### ISSUE TITLE
no theme folder is generated after running `yo liferay-theme:import`

### ISSUE THREAD

**Issue by john-co** [liferay/liferay-js-themes-toolkit#255]
Cannot find any imported theme after completing `yo liferay-theme:import` command. Discovered after #244 was resolved. Followed same steps in 244.

**Comment by wincent** [comment]
My understanding was that it was supposed to update the theme in-place, but perhaps that's mistaken. I'm not sure if these are the right docs to be looking at, but [this page](https://dev.liferay.com/develop/tutorials/-/knowledge_base/7-0/migrating-a-6-2-theme-to-liferay-7) says:

> The theme’s modified files (the files it modified from the base theme) are copied and reorganized in a newly created `src` directory.

My interpretation of that is that the new directory should be inside the existing theme folder. Is your understanding different @john-co?

**Comment by wincent** [comment]
Digging a little deeper, I do see in you test plan doc @john-co that it says "Assert zoe-healthcare-theme folder is generated".

I went back and installed v8.0.0-rc.3 of the toolkit and found that it doesn't create a new theme folder; it just updates in-place.

```
npm notice created a lockfile as package-lock.json. You should commit this file.
npm WARN manzanita-theme@1.0.0 No repository field.
npm WARN manzanita-theme@1.0.0 No license field.

added 1115 packages from 577 contributors and audited 10672 packages in 58.082s
found 55 vulnerabilities (17 low, 15 moderate, 23 high)
  run `npm audit fix` to fix them, or `npm audit` for details
events.js:173
      throw er; // Unhandled 'error' event
      ^

Error: spawn gulp ENOENT
 ```

I also installed v7.2.0 to see what it would do:

```
added 652 packages from 364 contributors and audited 6715 packages in 11.994s
found 57 vulnerabilities (18 low, 16 moderate, 23 high)
  run `npm audit fix` to fix them, or `npm audit` for details


   ╭────────────────────────────────────────────────────╮
   │                                                    │
   │   Update available 8.0.0-rc.3 → 8.0.2              │
   │   Run npm i -g generator-liferay-theme to update   │
   │                                                    │
   ╰────────────────────────────────────────────────────╯

events.js:173
      throw er; // Unhandled 'error' event
      ^

Error: spawn gulp ENOENT
```

🤦‍♂️ — so I think those errors are unfortunate but in both cases the error is happening at the end in the `install` (really the `gulp init`) step (and you can suppress the error by passing `--skip-install` to the generator) and the older versions of the toolkit did not create a new folder but just added a `src` folder to the existing theme, so I think this is a case of the behavior being expected and maybe our docs and/or test plan are out of date.

**Comment by wincent** [comment]
@john-co: given the analysis above, I'm going to close this but let me know if you think I've made a mistake and we can re-open.

**Comment by john-co** [comment]
@wincent based on [this page](https://dev.liferay.com/en/develop/tutorials/-/knowledge_base/7-0/migrating-a-6-2-theme-to-liferay-7) step 1 and [6.1 to 7.0 theme upgrade page](https://dev.liferay.com/en/develop/tutorials/-/knowledge_base/7-0/upgrading-your-theme-from-6-1-to-7-0#importing-your-theme) step no. 2 it says to `Navigate to the folder where you want your theme imported:` So I would assume the location that I run the next command `yo liferay-theme:import` would be where a new folder would be created and where the modified files would be copied into.

**Comment by john-co** [comment]
> Digging a little deeper, I do see in your test plan doc @john-co that it says "Assert zoe-healthcare-theme folder is generated".
> 
> I went back and installed v8.0.0-rc.3 of the toolkit and found that it doesn't create a new theme folder; it just updates in-place.
> 
> ```
> npm notice created a lockfile as package-lock.json. You should commit this file.
> npm WARN manzanita-theme@1.0.0 No repository field.
> npm WARN manzanita-theme@1.0.0 No license field.
> 
> added 1115 packages from 577 contributors and audited 10672 packages in 58.082s
> found 55 vulnerabilities (17 low, 15 moderate, 23 high)
>   run `npm audit fix` to fix them, or `npm audit` for details
> events.js:173
>       throw er; // Unhandled 'error' event
>       ^
> 
> Error: spawn gulp ENOENT
> ```
> I also installed v7.2.0 to see what it would do:
> 
> ```
> added 652 packages from 364 contributors and audited 6715 packages in 11.994s
> found 57 vulnerabilities (18 low, 16 moderate, 23 high)
>   run `npm audit fix` to fix them, or `npm audit` for details
> 
> 
>    ╭────────────────────────────────────────────────────╮
>    │                                                    │
>    │   Update available 8.0.0-rc.3 → 8.0.2              │
>    │   Run npm i -g generator-liferay-theme to update   │
>    │                                                    │
>    ╰────────────────────────────────────────────────────╯
> 
> events.js:173
>       throw er; // Unhandled 'error' event
>       ^
> 
> Error: spawn gulp ENOENT
> ```
>  — so I think those errors are unfortunate but in both cases the error is happening at the end in the `install` (really the `gulp init`) step (and you can suppress the error by passing `--skip-install` to the generator) and the older versions of the toolkit did not create a new folder but just added a `src` folder to the existing theme, so I think this is a case of the behavior being expected and maybe our docs and/or test plan are out of date.

I see, I'll update the test plan and assume the docs are out of date as well.

**Comment by john-co** [comment]
@wincent It does seem strange to me the use of the task name `import`. It may be misleading since it sounds like files are being created or downloaded but in reality it sounds more like an `update`

**Comment by wincent** [comment]
Sorry for the churn on this, but before you update it, I just wanted to check one more thing.

When I did my testing earlier I grabbed the theme and ran the generator in the same directory, so perhaps it is not surprising that it updated in place.

So just now I downloaded 7.2 of the toolkit and intentionally ran the import with the theme in a different directory, and I did get a different result:

```
❯ scratch
Spawing subshell in scratch directory:
  /var/folders/yc/1x2y2qld1g95tvtjpd89hdrm0000gp/T/tmp.zUTLRoC7
❯❯ cp ~/code/manzanita-theme-master.zip .
❯❯ unzip manzanita-theme-master.zip
❯❯ mv manzanita-theme{-master,}
❯❯ scratch
Spawing subshell in scratch directory:
  /var/folders/yc/1x2y2qld1g95tvtjpd89hdrm0000gp/T/tmp.DagBhQbQ
❯❯❯ npm install generator-liferay-theme@7.2.0
❯❯❯ yo ./node_modules/generator-liferay-theme/generators/import --skip-install

     _-----_
    |       |    .--------------------------.
    |--(o)--|    |  Welcome to the splendid |
   `---------´   |  Liferay Theme Importer  |
    ( _´U`_ )    |        generator!        |
    /___A___\    '--------------------------'
     |  ~  |
   __'.___.'__
 ´   `  |° ´ Y `

? What theme would you like to import? /var/folders/yc/1x2y2qld1g95tvtjpd89hdrm0000gp/T/tmp.zUTLRoC7/manzanita-theme
   Warning could not determine the property <template-extension> from /var/folders/yc/1x2y2qld1g95tvtjpd89hdrm0000gp/T/tmp.zUTLRoC7/manzanita-theme/docroot/WEB-INF/liferay-look-and-feel.xml. Using vm as default value.
❯❯❯ ls
manzanita-theme   node_modules      package-lock.json
```

So I'd hold off on updating the plan or the docs. We can and probably should update the tool instead.

> It does seem strange to me the use of the task name `import`. It may be misleading since it sounds like files are being created or downloaded but in reality it sounds more like an `update`.

I agree. It also seems weird to me that you would use a "generator" to do an "import", but 🤷‍♂️

**Comment by wincent** [comment]
And FWIW, I just replicated the same behavior with v8.0.0-rc.3.

Proof that QA saves lives!

**Comment by mwilliams2014** [comment]
Just seeing this thread now. The current instructions in the docs follow this set of steps:
1. navigate to the folder you want to import the 6.2 theme into
2. Run `yo liferay-theme:import`, and provide the absolute path to the 6.2 theme's root folder (I know you updated this @wincent to use relative paths now. Does that affect updating to 7.1 themes as well? If so we may need to update the 7.1 docs as well. The 7.2 docs are not written yet)
3. The theme's root folder is imported into the folder you ran the command in, and the files are updated to run with the toolkit

Is this what you guys expect @wincent @john-co ?

**Comment by john-co** [comment]
> Just seeing this thread now. The current instructions in the docs follow this set of steps:
> 
> 1. navigate to the folder you want to import the 6.2 theme into
> 2. Run `yo liferay-theme:import`, and provide the absolute path to the 6.2 theme's root folder (I know you updated this @wincent to use relative paths now. Does that affect updating to 7.1 themes as well? If so we may need to update the 7.1 docs as well. The 7.2 docs are not written yet)
> 3. The theme's root folder is imported into the folder you ran the command in, and the files are updated to run with the toolkit
> 
> Is this what you guys expect @wincent @john-co ?

Hi Michael. Thanks for verifying the instructions. Yes, this is what I would expect.

**Comment by wincent** [comment]
> I know you updated this @wincent to use relative paths now. Does that affect updating to 7.1 themes as well? Does that affect updating to 7.1 themes as well? If so we may need to update the 7.1 docs as well.

@mwilliams2014: it has no effect on `gulp update` or any other task, only on `yo liferay-theme:import`, which you can only run on 6.2 themes, and only with v8.x of the toolkit. Let me know if that's not clear.

**Comment by mwilliams2014** [comment]
@wincent that's clear. We'll just need to update our 7.1 docs to say the import accepts relative paths then. So, you're saying v 9.x.x of the generator will not support importing 6.2 themes?

**Comment by wincent** [comment]
> So, you're saying v 9.x.x of the generator will not support importing 6.2 themes, only 7.0 and up, I'm guessing?

*Almost*. It won't support importing 6.2 themes, and there is no such thing as "importing 7.0 themes" ("import" is a thing that only ever applied to 6.2 themes).

**Comment by mwilliams2014** [comment]
True @wincent . Alright. Thanks!

**Comment by wincent** [comment]
Just released [v8.0.3](https://github.com/liferay/liferay-js-themes-toolkit/releases/tag/v8.0.3) so marking this as closed.
